README.md

```diff
@@ -204,10 +204,10 @@ if bulb.is_variable_color_temp:
 
 ### Setting the color
 
-Hue is given in degrees (0-360) and saturation and value in percentage.
+Hue is given in degrees (0-360) and saturation and value in percentage. You can also add optional, fourth parameter - transition period (in milliseconds)
 
 ```python
 print(bulb.hsv)
 if bulb.is_color:
-   bulb.hsv = (180, 100, 100) # set to cyan
-```
\ No newline at end of file
+   bulb.hsv = (180, 100, 100, 1000) # set to cyan, fade 1 second
```

Comment:
I'm starting to dislike the property setter API, as this will make setting it ambiguous (as hsv stands for hue, saturation and value, and not for hue, saturation, value and transition). Would we be better off with providing `set_hsv(h, s, v, transition=500)` and trying to make people to convert to it?

Reply:
My point was to not set a new default, different than the bulb's inbuilt one. I do agree that overloading it is much better, and unless transition is specified to let the bulb handle the transition period. Maybe even create two setters - set_hsv() and set_hsvp() or set_hsvt() (p is for period, t is for transition, I like the latter better).

Reply:
We could just add `set_hsv()` which does the heavy lifting, and provide the property as an API to be used if one does not want to control the transition period. I don't see the value of adding an extra letter to the name.

Would that sound good to you? So something like this:

```
def set_hsv(hue, sat, val, transition=None):
    # do what is currently done in property setter here
    # apply transition if given etc.
```

and 
```
@hsv.setter
def hsv(hsv):
    self.set_hsv(hsv[0], hsv[1], hsv[2])
```
or similar?

Reply:
This is great 👍

Reply:
So, I think we should also then add setters for the rest of the functionality too, in case someone wants to use the non-property versions of them, right? That can be done in a separate PR though. @Jakpok, do you mind to update your PR accordingly?

Reply:
@rytilahti Ok, I can update my request, I will able to do it during the weekend